howso/hyperparameters.amlg

```diff
@@ -689,86 +657,46 @@
 		)
 		(call !ValidateParameters)
 
-		(declare (assoc
-			;local var
-			bad_param (false)
-		))
 
 		(if (!= (null) default_hyperparameter_map)
-			(if (or
-					(not (contains_index default_hyperparameter_map "p"))
-					(not (contains_index default_hyperparameter_map "k"))
-					(not (contains_index default_hyperparameter_map "dt"))
-				)
-				(assign (assoc bad_param (true) ))
-
-				(seq
-					(accum (assoc default_hyperparameter_map (assoc "paramPath" (list ".default")) ))
-					(assign_to_entities (assoc !defaultHyperparameters default_hyperparameter_map) )
-				)
+			(seq
+				(accum (assoc default_hyperparameter_map {"paramPath" [".default"]} ))
+				(assign_to_entities (assoc !defaultHyperparameters default_hyperparameter_map) )
 			)
 		)
 
 		;the passed in hyperparameters must at least have the basic attributes defined correctly
 		;iterate over action features
 		(if (!= (null) hyperparameter_map)
 			(let
-				(assoc param_paths (list))
+				(assoc
+					param_paths []
+					keys []
+				)
+
+				;populate all provided param_paths from the specified hyperparameter_map by appending all the indices recursively
+				#!RecursivelyAccumulateParamPaths
 				(map
```

Comment:
I think we should throw an (and) on this condition and also include dt and p. Mostly because I don't think it's too crazy to imagine a dataset with a feature named "k"

Reply:
Not to say there isn't one that could also have "p" and "dt", but certainly less likely.